### PR TITLE
[add] skin based icons

### PR DESCRIPTION
**Pull request by dentsu00**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -52,8 +52,8 @@ app.get("/folders", async (req, res, next)=>{
 	}));
 });
 
-app.get("/icons/:size/:type/:name", async (req, res, next)=>{
```
Comment (mkozhukh) [inline index.js:55]: It's fine, but lets preserve old routes as well ( for backward compatibility )
So calls to /icons/size/type/name will call getIconURL with empty skin value

File: index.js [hunk 2]
```diff
@@ -156,15 +156,22 @@ app.get("/direct", async (req, res, next) => {
...
+async function getIconURL(size, type, name, skin){
 	size = size.replace(/[^A-Za-z0-9.]/g, "");
 	name = name.replace(/[^A-Za-z0-9.]/g, "");
-	type - type.replace(/[^A-Za-z0-9.]/g, "");
+	type = type.replace(/[^A-Za-z0-9.]/g, "");
+	skin = skin.replace(/[^A-Za-z0-9.]/g, "");
 
 	name = "icons/" + size + "/" + name;
 
 	try {
```
Comment (mkozhukh) [inline index.js:171]: It will be nice to step "skin" folder testing step if skin value is empty

**Review by mkozhukh — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (dentsu00) [reply index.js:55]: done

File: index.js (hunk 2 above)
Comment (dentsu00) [reply index.js:171]: it won't be empty because the default is "none"

Comment (mkozhukh) [reply index.js:171]: And what is the reasoning of having such value?
When no-skin value provided, we need to return image from base set of images, without checking the skin folder ( as we want to preserve compatibility ) and if we are skipping check, using an empty value looks as a better choice ( as string value will not be used anywhere )

Comment (dentsu00) [reply index.js:171]: updated

**Pull request merged**